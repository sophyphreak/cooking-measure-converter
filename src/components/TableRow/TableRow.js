import React from 'react';
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  Input
} from 'reactstrap';

export default class TableRow extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      metricDropdownOpen: false,
      imperialDropdownOpen: false
    };

    this.toggleMetric = this.toggleMetric.bind(this);
    this.toggleImperial = this.toggleImperial.bind(this);
  }
  toggleMetric() {
    this.setState(prevState => ({
      metricDropdownOpen: !prevState.metricDropdownOpen
    }));
  }

  toggleImperial() {
    this.setState(prevState => ({
      imperialDropdownOpen: !prevState.imperialDropdownOpen
    }));
  }

  render() {
    const {
      metricDropdown,
      metricInput,
      imperialInput,
      imperialDropdown
    } = this.props;
    return (
      // TODO
      //
      // (1) Adapt for string label instead of dropdown button
      // (2) Break out a bunch of this into new components
      <tr>
        <th>
          <Dropdown
            isOpen={this.state.metricDropdownOpen}
            toggle={this.toggleMetric}
            value={metricDropdown.value}
          >
            <DropdownToggle caret>{metricDropdown.value}</DropdownToggle>
            <DropdownMenu>
              {metricDropdown.options.map((option, key) => {
                return (
                  <DropdownItem
                    key={key}
                    value={option}
                    onClick={() => metricDropdown.onChange(option)}
                  >
                    {option}
                  </DropdownItem>
                );
              })}
            </DropdownMenu>
          </Dropdown>
        </th>
        <td>
          <Input
            type="text"
            value={metricInput.value}
            onChange={metricInput.onChange}
          />
        </td>
        <td>
          <Input
            type="text"
            value={imperialInput.value}
            onChange={imperialInput.onChange}
          />
        </td>
        <th>
          <Dropdown
            isOpen={this.state.imperialDropdownOpen}
            toggle={this.toggleImperial}
            value={imperialDropdown.value}
          >
            <DropdownToggle caret>{imperialDropdown.value}</DropdownToggle>
            <DropdownMenu>
              {imperialDropdown.options.map((option, key) => {
                return (
                  <DropdownItem
                    key={key}
                    value={option}
                    onClick={() => imperialDropdown.onChange(option)}
                  >
                    {option}
                  </DropdownItem>
                );
              })}
            </DropdownMenu>
          </Dropdown>
        </th>
      </tr>
    );
  }
}
