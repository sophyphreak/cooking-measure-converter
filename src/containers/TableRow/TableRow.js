import React from 'react';
import TableRowComponent from '../../components/TableRowComponent/TableRowComponent';

export default class TableRow extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      metricDropdownOpen: false,
      imperialDropdownOpen: false
    };

    this.toggleMetric = this.toggleMetric.bind(this);
    this.toggleImperial = this.toggleImperial.bind(this);
  }
  toggleMetric() {
    this.setState(prevState => ({
      metricDropdownOpen: !prevState.metricDropdownOpen
    }));
  }

  toggleImperial() {
    this.setState(prevState => ({
      imperialDropdownOpen: !prevState.imperialDropdownOpen
    }));
  }

  render() {
    let {
      metricDropdown,
      metricInput,
      imperialInput,
      imperialDropdown
    } = this.props;
    metricDropdown.isOpen = this.state.metricDropdownOpen;
    metricDropdown.toggle = this.toggleMetric;
    imperialDropdown.isOpen = this.state.imperialDropdownOpen;
    imperialDropdown.toggle = this.toggleImperial;
    return (
      <TableRowComponent
        metricDropdown={metricDropdown}
        metricInput={metricInput}
        imperialInput={imperialInput}
        imperialDropdown={imperialDropdown}
      />
    );
  }
}
