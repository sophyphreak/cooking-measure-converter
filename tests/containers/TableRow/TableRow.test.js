import React from 'react';
import { shallow } from 'enzyme';
import TableRow from '../../../src/containers/TableRow/TableRow';

let wrapper, metricDropdown, metricInput, imperialInput, imperialDropdown;

beforeEach(() => {
  metricDropdown = {};
  metricInput = {};
  imperialInput = {};
  imperialDropdown = {};
  wrapper = shallow(
    <TableRow
      metricDropdown={metricDropdown}
      metricInput={metricInput}
      imperialInput={imperialInput}
      imperialDropdown={imperialDropdown}
    />
  );
});

test('should render TableRow correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
