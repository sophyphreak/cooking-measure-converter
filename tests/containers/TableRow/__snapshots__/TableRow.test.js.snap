// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TableRow correctly 1`] = `
<TableRowComponent
  imperialDropdown={
    Object {
      "isOpen": false,
      "toggle": [Function],
    }
  }
  imperialInput={Object {}}
  metricDropdown={
    Object {
      "isOpen": false,
      "toggle": [Function],
    }
  }
  metricInput={Object {}}
/>
`;
